drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -203,19 +205,19 @@ def self.find_definition_for_service_type(service_type, project:)
   end
 
   # Validate JSON definition when loading, to ensure no duplicate link IDs
-  def self.validate_json(json, valid_pick_lists: [])
+  def self.validate_json(json, valid_pick_lists: [], on_error: ->(err) { raise err })
```

Comment:
I haven't see this kind of pattern used in ruby much, it seems very JS-ish. Why not do a `rescue` to track the failures instead? We could define a specific error type on the Definition.

Reply:
I think rescue will only be hit for the first error, I want to collect ALL of the errors, not just the first one. Maybe a more ruby-ish pattern would be to `yield` errors, which would let the caller of the validate method decide how to handle them. That's basically what the aim is here, but that would be more ruby-ish I suppose.

Reply:
Adjusted this so that it will `yield` errors if a block is given, and will `raise` when it encounters an error if no block is given